src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/reroute/FlowRerouteFsm.java

```diff
@@ -466,6 +464,19 @@ public FlowRerouteFsm newInstance(@NonNull String flowId, @NonNull CommandContex
             });
             return fsm;
         }
+
+        private ErrorType extractErrorType(FlowRerouteFsm fsm) {
+            return Optional.ofNullable(fsm.getOperationResultMessage())
+                    .filter(message -> message instanceof ErrorMessage || message instanceof InfoMessage)
+                    .map(message -> {
+                        if (message instanceof InfoMessage) {
+                            return ErrorType.ALREADY_ON_BEST_PATH;
```

Comment:
Is this tested?

Reply:
thanks
cover with the unit tests all the corner cases